From 63a3fb0e9cd15e7f148a9190d2789270755b479b Mon Sep 17 00:00:00 2001
From: Bo-Cun Chen <bc-bocun.chen@mediatek.com>
Date: Fri, 3 Oct 2025 10:43:31 +0800
Subject: [PATCH] arm64: dts: mt7988: increase mdc for aqr 10gphy

Without this patch, the AQR 10G PHY driver might take too much time
to download the firmware.

Signed-off-by: Bo-Cun Chen <bc-bocun.chen@mediatek.com>
---
 arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-aqr.dtso     | 1 +
 arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-cux3410.dtso | 1 +
 arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-aqr.dtso     | 1 +
 arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-cux3410.dtso | 1 +
 arch/arm64/boot/dts/mediatek/mt7988a-rfb.dts               | 5 -----
 5 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-aqr.dtso b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-aqr.dtso
index a9c3101..ee37c1b 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-aqr.dtso
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-aqr.dtso
@@ -17,6 +17,7 @@ fragment@0 {
 		__overlay__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
+			clock-frequency = <10500000>;
 
 			/* external Aquantia AQR113C */
 			phy0: ethernet-phy@0 {
diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-cux3410.dtso b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-cux3410.dtso
index 9ee1a4c..72491db 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-cux3410.dtso
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-cux3410.dtso
@@ -18,6 +18,7 @@ fragment@0 {
 		__overlay__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
+			clock-frequency = <10500000>;
 
 			/* external Aquantia CUX3410 */
 			phy0: ethernet-phy@0 {
diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-aqr.dtso b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-aqr.dtso
index 6563800..1b687ea 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-aqr.dtso
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-aqr.dtso
@@ -17,6 +17,7 @@ fragment@0 {
 		__overlay__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
+			clock-frequency = <10500000>;
 
 			/* external Aquantia AQR113C */
 			phy8: ethernet-phy@8 {
diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-cux3410.dtso b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-cux3410.dtso
index b53300c..970521a 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-cux3410.dtso
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-cux3410.dtso
@@ -17,6 +17,7 @@ fragment@0 {
 		__overlay__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
+			clock-frequency = <10500000>;
 
 			/* external Aquantia CUX3410 */
 			phy8: ethernet-phy@8 {
diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-rfb.dts b/arch/arm64/boot/dts/mediatek/mt7988a-rfb.dts
index 2a798c9..fca9a3d 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-rfb.dts
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-rfb.dts
@@ -50,11 +50,6 @@ trng{
 	};
 };
 
-&eth {
-	pinctrl-0 = <&mdio0_pins>;
-	pinctrl-names = "default";
-};
-
 &gmac0 {
 	status = "okay";
 };
-- 
2.45.2

